pkg/api/controller.go

```diff
@@ -1686,6 +1686,28 @@ func (c *Controller) UploadObject(w http.ResponseWriter, r *http.Request, reposi
 		return
 	}
 
+	// before writing body, ensure preconditions - this means we essentially check for object existence twice:
+	// once before uploading the body to save resources and time,
+	//	and then graveler will check again when passed a WriteCondition.
+	allowOverwrite := true
```

Comment:
1. Swagger validator should catch invalid value for `IfNoneMatch` - part of the JSON schema validation (if not ignore this comment)
2. Why do we need to write code at this level to verify the existence and not just pass the boolean value to the underlying catalog? if WriteCondition members were public this will be easier to pass in this case.

Reply:
The idea is to test this specific condition outside of Graveler - why wait for the user to upload the body only to fail at CreateEntry if I already know this write is going to fail?